Frameworks/AudioToolbox/AudioConverter.mm

```diff
@@ -16,28 +17,186 @@
  @Notes
 */
 OSStatus AudioConverterDispose(AudioConverterRef inAudioConverter) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    HRESULT status;
+
+    status = MFShutdown();
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"MFShutdown Failed");
+        return kAudioConverterErr_UnspecifiedError;
+    }
+
+    CoUninitialize();
+    delete inAudioConverter;
+
+    return noErr;
+}
+
+void _setMFProperties(const AudioStreamBasicDescription* format, IMFMediaType** mediaType) {
+    IMFMediaType* tempMediaType = nullptr;
+    MFCreateMediaType(&tempMediaType);
+
+    tempMediaType->SetGUID(MF_MT_MAJOR_TYPE, MFMediaType_Audio);
+
+    if (format->mFormatFlags & kAudioFormatFlagIsFloat) {
+        tempMediaType->SetGUID(MF_MT_SUBTYPE, MFAudioFormat_Float);
+    } else {
+        tempMediaType->SetGUID(MF_MT_SUBTYPE, MFAudioFormat_PCM);
+    }
+
+    tempMediaType->SetUINT32(MF_MT_AUDIO_NUM_CHANNELS, format->mChannelsPerFrame);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_SAMPLES_PER_SECOND, format->mSampleRate);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_BLOCK_ALIGNMENT, format->mBytesPerFrame);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_AVG_BYTES_PER_SECOND, format->mSampleRate * format->mBytesPerFrame);
+    tempMediaType->SetUINT32(MF_MT_AUDIO_BITS_PER_SAMPLE, format->mBitsPerChannel);
+    tempMediaType->SetUINT32(MF_MT_ALL_SAMPLES_INDEPENDENT, TRUE);
+
+    *mediaType = tempMediaType;
 }
 
 /**
- @Status Stub
+@Status Caveat: AudioConverter supported only for conversions through AudioConverterConvertBuffer.
+                Only Linear PCM formats are supported with 1 Frame per Packet and equal source & destination Channels per Frame.
+                Returns kAudioConverterErr_UnspecifiedError on failure and noErr (0) on success. Other return types not supported.
 */
 OSStatus AudioConverterNew(const AudioStreamBasicDescription* inSourceFormat,
                            const AudioStreamBasicDescription* inDestinationFormat,
-                           DWORD* outAudioConverter) {
-    // TODO: outAudioConverter ought to be made AudioConverterRef _Nullable * when this is unstubbed
-    UNIMPLEMENTED();
-    if (outAudioConverter) {
-        *outAudioConverter = 1234;
+                           AudioConverterRef _Nullable* outAudioConverter) {
+    if (inSourceFormat->mFormatID != kAudioFormatLinearPCM || inDestinationFormat->mFormatID != kAudioFormatLinearPCM) {
+        UNIMPLEMENTED();
+        return StubReturn();
     }
-    return StubReturn();
+
+    if (inSourceFormat->mChannelsPerFrame != inDestinationFormat->mChannelsPerFrame) {
+        UNIMPLEMENTED();
+        return StubReturn();
+    }
+
+    if (inSourceFormat->mFramesPerPacket != 1 || inSourceFormat->mFramesPerPacket != inDestinationFormat->mFramesPerPacket) {
+        UNIMPLEMENTED();
+        return StubReturn();
+    }
+
+    HRESULT status;
+
+    status = CoInitializeEx(nullptr, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
```

Comment:
>CoInitializeEx [](start = 13, length = 14)

is this needed? I think we CoInit on our threads already and I'm not very comfortable with the flag you have here. Can you explain the logic behind this?